roles/mysql/tasks/main.yml

```diff
@@ -18,3 +18,4 @@
     password: "{{ db_password }}"
     state: present
     priv: "{{ competition_name }}.*:ALL"
+    append_privs: yes
```

Comment:
Similarly to above (though it's fine to just have the explanation in one place and refer to it from other places, of course).

Reply:
Is this comment missing context?  (I assume you had a comment asking for an explanation of what this change was doing?)

Reply:
Yeah, I don't know where my previous comment went.  I definitely included it in my original review.

Anyway, what it said was that the commit that added the `append_privs` lines included, in the commit message, an explanation of what the lines do.  And I was just saying instead of putting that explanation in the commit message, put it into a comment(s) in the code itself.

Reply:
Got it, can / will do -- but just because why not ask: I always thought of commit messages as the right place for that kinda context because it's less "in your face" but still just a quick git blame away from anybody who is interested in the background / intention behind a given line.  (I appreciate that by asking I'm nearing bike shedding but without asking / using at least a little paint I'll never adjust my instinct)

Reply:
This sparked a discussion offline among some other devs and the thought is: change context belongs in commit; which isn't the same as explaining what a line actually does.

Given that I have been thinking about this more and I'm torn because (A) folks could (should?) refer to Ansible documentation to know what append_privs does and also (B) if I were to document it would be like `append_privs: yes // adds to privilieges instead of overwriting them` which feels almost redundant since `append_privs` is a fairly clear setting name to begin with.

(don't worry: I'm not spending as much time on this as it appears! This is all background processing stuff as I work on other things)

Reply:
Heh.  Well, let's separate out the two unrelated questions here:

1) What should we assume a reader already has as part of their default background knowledge?

There's no one right answer to this.  Some readers will know Ansible well; others less well; still others barely at all.  It's purely a judgement call.  Within OTS, most of us are Ansible amateurs, so a comment like `// adds to privileges instead of overwriting them` helps -- it's not completely obvious from the name `append_privs` that that's what's going on.  But I think you'd be on solid ground leaving it out, too, for the reasons you gave.

2) What should go in commit messages vs in code comments or other in-diff documentation?

The main purpose of a commit message is to put a reader in the right frame of mind to understand the diff as a whole -- to explain the motivation and overall shape of the change, so that the reader's expectations are set properly for absorbing the details of the diff.  Thus, almost always, when there is any question of whether a bit of technical explanation should go in the commit message or in the code, the answer is that it should go in the code.  Normally, people will be reading the code without having related commit messages easily at hand, so if there's something specific they should know at the line level -- just to understand the code (as opposed to the code *change*) in front of them -- they'll want the explanation right there.

Shorthand rule-of-thumb: comments are about the *code*, commit messages are about the *change*.

And, I think this is exactly what you were saying in your second comment, so I'm just re-articulating something back at you that you were already on board with.  What can I say?  I'll be here all week; please tip your waiters, folks.